test/unit/util/vm-manager-hoc.test.jsx

```diff
@@ -75,27 +76,38 @@ describe('VMManagerHOC', () => {
         // nextTick needed since vm.loadProject is async, and we have to wait for it :/
         process.nextTick(() => expect(mockedOnLoadedProject).toHaveBeenLastCalledWith(LoadingState.LOADING_VM_WITH_ID));
     });
-    test('if there is projectData, the child is rendered', () => {
+    test('if the fontsLoaded prop becomes true, it loads project data into the vm', () => {
+        vm.loadProject = jest.fn(() => Promise.resolve());
+        const mockedOnLoadedProject = jest.fn();
         const Component = () => <div />;
         const WrappedComponent = vmManagerHOC(Component);
         const mounted = mount(
             <WrappedComponent
-                projectData="100"
+                isLoadingWithId
                 store={store}
                 vm={vm}
+                onLoadedProject={mockedOnLoadedProject}
             />
         );
-        expect(mounted.find('div').length).toBe(1);
+        mounted.setProps({
+            fontsLoaded: true,
+            loadingState: LoadingState.LOADING_VM_WITH_ID,
+            projectData: '100'
+        });
+        expect(vm.loadProject).toHaveBeenLastCalledWith('100');
+        // nextTick needed since vm.loadProject is async, and we have to wait for it :/
+        process.nextTick(() => expect(mockedOnLoadedProject).toHaveBeenLastCalledWith(LoadingState.LOADING_VM_WITH_ID));
     });
-    test('if there is no projectData, nothing is rendered', () => {
+    test('if there is projectData, the child is rendered', () => {
```

Comment:
I'm not sure this test makes sense anymore. Seems like now, if there isn't projectData, the child is still rendered.

Reply:
Sure, I can take this test out too

It merged my tests in a weird way. Basically I took out the nothing is rendered test and added a test that load happens when fontsLoaded becomes true. Should there be another test where fontsLoaded doesn't become true, and then vm.loadProject is never called?

Reply:
The problem is there's no real way (that I know) to test "is never called"

Reply:
I took out the child rendered test and added a doesn't-load-without-fonts test. I think .toHaveBeenCalledTimes(0) works